pkg/kubectl/sorting_printer.go

```diff
@@ -60,8 +61,28 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
 	if len(objs) == 0 {
 		return nil
 	}
+
+	if err := SortObjects(objs, s.SortField); err != nil {
+		return err
+	}
+
+	switch list := obj.(type) {
+	case *v1.List:
+		for ix := range objs {
+			data, err := v1.Codec.Encode(objs[ix])
```

Comment:
This is non-obvious to me.  Lists can contain objects done in different codecs. Why force them out of their native shape?